sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java

```diff
@@ -815,90 +1023,186 @@ private static class RntbdMetricsV2 implements RntbdMetricsCompletionRecorder {
         private final Timer responseErrors;
         private final DistributionSummary responseSize;
         private final Timer responseSuccesses;
+        private final EnumSet<MetricCategory> metricCategories;
 
         private RntbdMetricsV2(MeterRegistry registry, RntbdTransportClient client, RntbdEndpoint endpoint) {
             Tags tags = Tags.of(endpoint.clientMetricTag());
+            this.metricCategories = client.getMetricCategories();
+            if (metricCategories.contains(MetricCategory.DirectRequests)) {
+
+                CosmosMeterOptions options = client
+                    .getMeterOptions(CosmosMeterName.DIRECT_REQUEST_LATENCY);
+                if (options.isEnabled()) {
+                    this.requests = Timer
+                        .builder(options.getMeterName().toString())
+                        .description("RNTBD request latency")
+                        .maximumExpectedValue(Duration.ofSeconds(300))
+                        .publishPercentiles(optionsAccessor.getPercentiles(options))
+                        .publishPercentileHistogram(optionsAccessor.isHistogramPublishingEnabled(options))
+                        .tags(getEffectiveTags(tags, options))
+                        .register(registry);
+                } else {
+                    this.requests = null;
```

Comment:
Since these are created in the constructor, how it will react to the dynamic option modification? (for example, when I remove or add a new metrics category, will it reevaluate requests timer etc?)